src/sentry/notifications/notifications/activity/base.py

```diff
@@ -50,7 +51,8 @@ def get_recipient_context(
         context = super().get_recipient_context(recipient, extra_context)
         return {**context, **get_reason_context(context)}
 
-    def get_reference(self) -> Any:
+    @property
+    def reference(self) -> Model | None:
```

Comment:
@mgaeta shouldn't the return type here be the model class for an activity? Is it possible for the return value to be `None` for this subclass?

Reply:
Because `reference()` is overriding an abstract method, the signatures have to match for the interface to be valid. The options for `reference` values are `Organization`, `Group`, `Project`, `Activity`, or `None` so the best option is `Model | None`.

Reply:
Damn, I guess Python isn't as smart as TS in allowing a more restrictive signature